tests/types/test-array.py

```diff
@@ -190,3 +190,61 @@ def test_create_from_python_array_incompatible_child_types():
           "<class 'str'>, whereas previous elements were int32"
     with pytest.raises(TypeError, match=msg):
         DT = dt.Frame(E=src)
+
+
+
+#-------------------------------------------------------------------------------
+# Convert to (any)
+#-------------------------------------------------------------------------------
+
+def test_arr32_repr_in_terminal():
+    DT = dt.Frame(A=[[1], [2, 3], None, [4, 5, 6], []])
+    assert str(DT) == (
+        "   | A           \n"
+        "   | arr32(int32)\n"
+        "-- + ------------\n"
+        " 0 | [1]         \n"
+        " 1 | [2, 3]      \n"
+        " 2 | NA          \n"
+        " 3 | [4, 5, 6]   \n"
+        " 4 | []          \n"
+        "[5 rows x 1 column]\n"
+    )
+
+
+def test_arr32_repr_in_terminal2():
+    DT = dt.Frame(A=[['abeerfaer'] * 30])
+    assert str(DT) == (
+        "   | A                                                                                                   \n"
+        "   | arr32(str32)                                                                                        \n"
+        "-- + ----------------------------------------------------------------------------------------------------\n"
+        " 0 | [abeerfaer, abeerfaer, abeerfaer, abeerfaer, abeerfaer, abeerfaer, abeerfaer, abeerfaer, abeerf…, …]\n"
+        "[1 row x 1 column]\n"
+    )
+
+
+def test_arr32_arr32_repr():
+    DT = dt.Frame(V=[[[1, 2, 3], [4, 9]], None, [None], [[-1], [0, 13]]])
+    assert str(DT) == (
+        "   | V                  \n"
+        "   | arr32(arr32(int32))\n"
+        "-- + -------------------\n"
+        " 0 | [[1, 2, 3], [4, 9]]\n"
+        " 1 | NA                 \n"
+        " 2 | [NA]               \n"
+        " 3 | [[-1], [0, 13]]    \n"
+        "[4 rows x 1 column]\n"
+    )
+
+
+def test_arr32_of_strings_repr():
+    DT = dt.Frame(W=[['ad', 'dfkvjn'], ['b b, f', None], ['r', 'w', 'dfvdf']])
+    assert str(DT) == (
+        "   | W            \n"
+        "   | arr32(str32) \n"
+        "-- + -------------\n"
+        " 0 | [ad, dfkvjn] \n"
```

Comment:
I understand that when we print content of a string column we don't add quotes, because we are kind of printing the raw data. If one needs to copy and paste a string somewhere into the python code, it is not a big deal to add quotes.

However, in the case of printing `arr*` it could be a little bit confusing, because we stick to the python's list syntax and a user may think that copy-pasting a value into python may work. But, quotes should be added around each element and that could be boring.

Reply:
The discrepancy is more than just quotation marks. NA elements in the list will get rendered as `NA`, not `None` (which I think is good), the list itself will get shortened with an ellipsis `...` if there are too many elements, the individual elements in the list will get truncated too if they are too long -- see the test below for such a use case.

Obviously, the fact that we don't have quotes for strings here is somewhat problematic. For example, a string value may contain embedded `,` or square brackets, or even spaces -- which will make the output somewhat confusing. However, strings are not the only use-case. What if it's an array of dates? Right now it'll look as `[2000-01-01, 2001-02-22]`, but that is certainly not a proper python syntax either.

I'm not sure what the best approach should be here. Special handling for strings only? However, it won't be as simple as surrounding each element with quotation marks -- you also need to escape the quotes inside the string.

In any case, this seems like a very minor issue that can be postponed to some future time (or filed as "Beginner task" for the potential newcomers).

Reply:
May be we could add a Frame method called `.to_python()` or something that will return a valid python list of lists.

Reply:
We already have one: `.to_list()`. Alternatively, `.to_dict()` or `.to_tuples()`.